backend/inspirehep/disambiguation/cli.py

```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+import click
+from elasticsearch_dsl import Q
+from inspire_utils.record import get_values_for_schema
+from invenio_db import db
+
+from inspirehep.records.api import AuthorsRecord
+from inspirehep.records.models import RecordsAuthors
+from inspirehep.search.api import AuthorsSearch
+
+
+@click.group()
+def disambiguation():
+    """Commands for disambiguation"""
+
+
+@disambiguation.command()
+def clean_stub_authors():
+    stub_authors_query = Q("term", stub=True)
+    stub_authors = (
+        AuthorsSearch().query(stub_authors_query).source(["control_number"]).execute()
+    )
+    stub_authors_control_numbers = [
+        ("aut", str(author["control_number"])) for author in stub_authors
+    ]
+    stub_authors_db = list(
+        AuthorsRecord.get_records_by_pids(stub_authors_control_numbers)
+    )
+    db.session.connection(execution_options={"isolation_level": "SERIALIZABLE"})
+    for author in stub_authors_db:
```

Comment:
here you're doing lots of queries, one for each author, which is expensive. It's better to filter the stub authors first, then do a join against the `RecordsAuthors` table (potentially batched to avoid too long statements, look at how `get_records_by_pids` is implemented).